### ISSUE TITLE
[Bug] GeoDa raises wxwidgets debug alert "wxMenutItem::Check" on windows

### ISSUE THREAD

**Issue by nicoman** [GeoDaCenter/geoda#2458]
**Describe the bug**
Installed the new version 1.22.0.2 and while opening an old shape file got a long error starting with "A debugging check in this application has failed." ..\..\src\msw\menuitem.cpp(613): asset ""Assert failure"" failed in wxMenuItem::Check(): CheckMenuItem() failed, item not in the menu?

**To Reproduce**
1. Open\ESRI Shape file
2. Click on USStates shape file

**Expected behavior**
Just to open an existing shape file.

**Screenshots**
![geoda_error](https://github.com/GeoDaCenter/geoda/assets/29689971/a7357461-ea25-4b16-b81e-b6606a924c42)

**Desktop (please complete the following information):**
 - OS: Windows
 - Version win10

**Additional context**

**Comment by Jenhka** [comment]
Hello everyone! Was anyone able to solve this problem? This error keeps emerging upon opening any file and trying to perform any type of geospatial analysis, sometimes it's random, sometimes it's consistent. Would be really grateful if someone indicated a solution to this problem.

**Comment by lixun910** [comment]
I think it should be in version 1.22.0.14 Do you use the latest version? http://geodacenter.github.io/download.html. Thanks!

**Comment by Jenhka** [comment]
> I think it should be in version 1.22.0.14 Do you use the latest version? http://geodacenter.github.io/download.html. Thanks!

Yes, I use the latest version. Didn't do anything extra, just installed it. Tried to uninstall and reinstall it. However, the problem persists.

**Comment by lixun910** [comment]
I just tried opening a ESRI shapefile on my windows10 machine, and it works fine (see the screenshot). Are there any other steps to reproduce? Thanks!

<img width="616" alt="Image" src="https://github.com/user-attachments/assets/cee37200-cb78-4ec7-bc28-774d4482e82a" />

**Comment by Jenhka** [comment]
![Image](https://github.com/user-attachments/assets/e2dbfa5a-e0de-4758-97a0-7fd55c40e733)

**Comment by lixun910** [comment]
Thanks for sharing the dataset! I will try it on my side to replicate. Will keep you updated.

**Comment by lixun910** [comment]
@Jenhka I've tried to fix this bug and here is a new build for Windows: https://github.com/GeoDaCenter/geoda/actions/runs/14866305626/artifacts/3072456701
Can you help to test if this new build fixes this issue? Thanks very much!

**Comment by Jenhka** [comment]
@lixun910 Seems like the error stopped appearing when opening a shapefile, but after I created a weights matrix and attempted to open the Connectivity Graph it still pops up. When I press Continue I can still use GeoDa.

![Image](https://github.com/user-attachments/assets/39df60de-6278-4f1f-87be-941ba65fe496)

Haven't worked with the new build yet, but could this error screw up the results of spatial analysis? If it doesn't alter the results, then it's not a big deal for me.

**Comment by lixun910** [comment]
@Jenhka The error is a UI related error and it does NOT impact the spatial analysis results. The new build just contains a fix for this UI error, so please let me know if the new build works when you have time. Thanks again!

**Comment by lixun910** [comment]
I think I find the bug that causes the error. Here is a new build that you can download from [here](https://productionresultssa7.blob.core.windows.net/actions-results/f2e5880a-3e62-4c0a-bc27-4c2acd07eb8f/workflow-job-run-8b10539f-4d5a-5bc6-8741-980a592fe859/artifacts/bef9ab08a7fbd14e1e0f45c049a1c0eb1846985b623d07b1c834cf07c1642e76.zip?rscd=attachment%3B+filename%3D%22GeoDa-Windows7%2B-x64-installer.zip%22&se=2025-05-09T17%3A07%3A58Z&sig=c21dFx6n650Ydhzg3N05mjLko3%2BAiYq4lyig1CdG1WQ%3D&ske=2025-05-10T01%3A02%3A42Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2025-05-09T13%3A02%3A42Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-05-05&sp=r&spr=https&sr=b&st=2025-05-09T16%3A57%3A53Z&sv=2025-05-05) or https://github.com/GeoDaCenter/geoda/actions/runs/14916329290/artifacts/3089821466 
Please let me know if the Connectivity Graph works fine without poping up the Debug window in this new build. Thank you, @Jenhka !

**Comment by lixun910** [comment]
Sorry, the download link I just posted is not valid. Here is the valid download link: https://github.com/GeoDaCenter/geoda/actions/runs/14934143205/artifacts/3095747519

**Comment by Jenhka** [comment]
The Connectivity Graph works fine, but the same error keeps randomly popping up upon doing other actions, for example clustering, Moran's, etc. Maybe it has to do with something on my end?

**Comment by lixun910** [comment]
I think it may involves all the menu actions (not only wxMenuItem::check. I will make another fix to all possible menu actions. Will keep you posted. Thanks for helping fixing this issue!

**Comment by lixun910** [comment]
Yes, here is a new build with a fix for this issue: https://github.com/GeoDaCenter/geoda/actions/runs/15522763577/artifacts/3284472782

Let me know if this build fixes the issue! :)

**Comment by lixun910** [comment]
@Jenhka Do you have a chance to try the latest build? Thanks!

**Comment by Jenhka** [comment]
I just tried it and I'm afraid it still has the same error. At first it worked fine until I clicked on the Connectivity Map after creating a weights matrix. The second time it occurred after I closed and opened GeoDa again.
Like I assumed before, is there a chance there's something wrong on my side? But I don't know what it might be then...
